File: src/KeePassLib2Android/PwDeletedObject.cs
/*
  KeePass Password Safe - The Open-Source Password Manager
  Copyright (C) 2003-2012 Dominik Reichl <dominik.reichl@t-online.de>

  This program is free software; you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation; either version 2 of the License, or
  (at your option) any later version.

  This program is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU General Public License for more details.

  You should have received a copy of the GNU General Public License
  along with this program; if not, write to the Free Software
  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
*/

using System;
using System.Collections.Generic;
using System.Diagnostics;

using KeePassLib.Interfaces;

namespace KeePassLib
{
	/// <summary>
	/// Represents an object that has been deleted.
	/// </summary>
	public sealed class PwDeletedObject : IDeepCloneable<PwDeletedObject>
	{
		private PwUuid m_uuid = PwUuid.Zero;
		/// <summary>
		/// UUID of the entry that has been deleted.
		/// </summary>
		public PwUuid Uuid
		{
			get { return m_uuid; }
			set
			{
				if(value == null) throw new ArgumentNullException("value");
				m_uuid = value;
			}
		}

		private DateTime m_dtDeletionTime = PwDefs.DtDefaultNow;
		/// <summary>
		/// The date/time when the entry has been deleted.
		/// </summary>
		public DateTime DeletionTime
		{
			get { return m_dtDeletionTime; }
			set { m_dtDeletionTime = value; }
		}

		/// <summary>
		/// Construct a new <c>PwDeletedObject</c> object.
		/// </summary>
		public PwDeletedObject()
		{
		}

		public PwDeletedObject(PwUuid uuid, DateTime dtDeletionTime)
		{
			if(uuid == null) throw new ArgumentNullException("uuid");

			m_uuid = uuid;
			m_dtDeletionTime = dtDeletionTime;
		}

		/// <summary>
		/// Clone the object.
		/// </summary>
		/// <returns>Value copy of the current object.</returns>
		public PwDeletedObject CloneDeep()
		{
			PwDeletedObject pdo = new PwDeletedObject();

			pdo.m_uuid = m_uuid; // PwUuid objects are immutable
			pdo.m_dtDeletionTime = m_dtDeletionTime;

			return pdo;
		}
	}
}

File: src/java/KP2AKdbLibrary/app/src/main/java/com/keepassdroid/database/PwDeletedObject.java
/*
 * Copyright 2010-2013 Brian Pellin.
 *     
 * This file is part of KeePassDroid.
 *
 *  KeePassDroid is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 2 of the License, or
 *  (at your option) any later version.
 *
 *  KeePassDroid is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with KeePassDroid.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
package com.keepassdroid.database;

import java.util.Date;
import java.util.UUID;

public class PwDeletedObject {
	public UUID uuid;
	private Date deletionTime;
	
	public PwDeletedObject() {
		
	}
	
	public PwDeletedObject(UUID u) {
		this(u, new Date());
	}
	
	public PwDeletedObject(UUID u, Date d) {
		uuid = u;
		deletionTime = d;
	}
	
	public Date getDeletionTime() {
		if ( deletionTime == null ) {
			return new Date(System.currentTimeMillis());
		}
		
		return deletionTime;
	}
	
	public void setDeletionTime(Date date) {
		deletionTime = date;
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) {
			return true;
		}
		else if (o == null) {
			return false;
		}
		else if (!(o instanceof PwDeletedObject)) {
			return false;
		}
		
		PwDeletedObject rhs = (PwDeletedObject) o;
		
		return uuid.equals(rhs.uuid);
	}
}
